### ISSUE TITLE
[Test] iterator_test_template: add sized_sentinel property

### ISSUE THREAD

**Issue by marehr** [seqan/product_backlog#192]
### Description 

We currently have:

```c++

    static constexpr bool const_iterable = true/false;         // Also test const_iterability. (const begin/end required)
```

https://github.com/seqan/seqan3/blob/323b8252514855700ea8b2e701d38a5e9e46e6cc/test/unit/range/iterator_test_template.hpp#L28-L30

This allows to say that an iterator has the const_iterable property. That means that `std::ranges::cbegin(range)` is a valid operation.

Another valid operation for some iterator/sentinel is `iterator - iterator` (if `std::sized_sentinel_for<iterator, iterator>`), `sentinel - iterator` or `iterator - sentinel` (if `std::sized_sentinel_for<iterator, sentinel>`).

```c++
static constexpr bool sized_iterator = true/false;
static constexpr bool sized_sentinel = true/false;
```

We already have two tests `difference_common` and `difference_sentinel` within the test template that test exactly this, but they will only be executed if the iterator is at least a random_access_iterator (every random_access_iterator models `sized_iterator`, but might not model `sized_sentinel` e.g. unbounded repeat view / iota view).

It is also possible that forward or bidirectional iterator fulfil this property, like pseudo_random_access_iterator do (i.e. any tree like structure normally does this).

### Acceptance Criteria

- [ ] test 1
- [ ] test 2
- [ ] test 3

### Tasks

 

- [ ] task 1
- [ ] task 2
- [ ] task 3

### Definition of Done

- [ ] Implementation and design approved
- [ ] Unit tests pass
- [ ] Test coverage = 100%
- [ ] Microbenchmarks added and/or affected microbenchmarks < 5% performance drop
- [ ] API documentation added
- [ ] Tutorial/teaching material added
- [ ] Test suite compiles in less than 30 seconds (on travis)
- [ ] Changelog entry added

**Comment by rrahn** [comment]
> every random_access_iterator fulfils sized_iterator and sized_sentinel).

Is this not conditionally enabled? If they are not common ranges then they might not model sized_sentinel_for<iterator_t, sentinel_t> even if they model random_access_iterator, e.g. the unbounded repeat view.

**Comment by marehr** [comment]
> > every random_access_iterator fulfils sized_iterator and sized_sentinel).
> 
> Is this not conditionally enabled? If they are not common ranges then they might not model sized_sentinel_for<iterator_t, sentinel_t> even if they model random_access_iterator, e.g. the unbounded repeat view.

You are right, I always forget about those weird cases, but as you can see, sentinel_difference is ALWAYS enabled for >=random_access_iterator_tag.

```c++
    if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::random_access_iterator_tag>)
    {
        using difference_t = std::ranges::range_difference_t<decltype(this->test_range)>;


        auto && begin = std::ranges::begin(this->test_range);
        auto && end = std::ranges::end(this->test_range);
        difference_t size = std::ranges::distance(this->expected_range);


        EXPECT_EQ(size, end - begin);
        EXPECT_EQ(-size, begin - end);
```
https://github.com/seqan/seqan3/blob/c24c911880d350e3388616563ad73b9ca9ac0ba7/test/unit/range/iterator_test_template.hpp#L463-L472
